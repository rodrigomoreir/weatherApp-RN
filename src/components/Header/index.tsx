import React from 'react';

import {
  StyledContainer,
  StyledTitle,
  StyledImageContainer,
  StyledImage,
  StyledContent,
  StyledTitleWithGoBack
} from './styles';

import chevronLeft from '../../assets/icons/chevronLeft.png'

interface Props {
  title: string;
  goBack?: () => void;
}

export const Header = ({ title, goBack }: Props) => {
  return (
    <StyledContainer>
      {goBack ? (
        <StyledContent>
          <StyledImageContainer onPress={goBack}>
            <StyledImage source={chevronLeft} />
          </StyledImageContainer>
          <StyledTitleWithGoBack>{title}</StyledTitleWithGoBack>
        </StyledContent>
      ) :
        <StyledTitle>{title}</StyledTitle>
      }
    </StyledContainer >
  );
}
