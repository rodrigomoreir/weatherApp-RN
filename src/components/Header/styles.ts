import styled from 'styled-components/native';

export const StyledContainer = styled.View`
  background-color: ${({ theme }) => theme.colors.background};
  padding: 15px 15px;
`;

export const StyledTitle = styled.Text`
  font-size: 25px;
  font-weight: bold;
  margin-bottom: 10px;
  color: ${({ theme }) => theme.colors.textPrimary};
`;

export const StyledContent = styled.View`
  flex-direction: row;
  align-items: center;
`

export const StyledTitleWithGoBack = styled.Text`
  font-size: 25px;
  font-weight: bold;
  margin-left: 20px;
  color: ${({ theme }) => theme.colors.textPrimary};
`;

export const StyledImageContainer = styled.TouchableOpacity`
  background-color: ${({ theme }) => theme.colors.card};
  width: 40px;
  height: 40px;
  align-items: center;
  justify-content: center;
  border-radius: 20px;
  border-width: 0.5px;
  border-color: ${({ theme }) => theme.colors.border}
`
export const StyledImage = styled.Image`
  height: 20px;
  width: 20px;
  tint-color: ${({ theme }) => theme.colors.border}
`
