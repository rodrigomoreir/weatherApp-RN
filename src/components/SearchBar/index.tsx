import React from 'react';
import { Image } from 'react-native';

import icSearch from '../../assets/icons/icSearch.png';

import {
  StyledContainer,
  StyledTextInput,
  StyledIconContainer
} from './styles';

interface Props {
  placeholder: string;
  value: string;
  onChangeText: () => void;
}

const SearchBar = ({ placeholder, value, onChangeText }: Props) => {
  return (
    <StyledContainer>
      <StyledTextInput
        testID='text-input'
        placeholder={placeholder}
        value={value}
        onChangeText={onChangeText}
      />
      <StyledIconContainer testID='pressable-icon'>
        <Image source={icSearch} />
      </StyledIconContainer>
    </StyledContainer>
  );
};

export default SearchBar;
