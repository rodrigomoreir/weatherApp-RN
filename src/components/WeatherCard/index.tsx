import React from 'react';
import { Text } from 'react-native';

import {
  StyledContainer,
  StyledLeftContent,
  StyledRightContent,
  StyledCityContent,
  StyledTitle,
  StyledSubtitle,
  StyledTemperature
} from './styles';

interface Props {
  city: string;
  hour: string;
  weather: string;
  temperature: string;
  tempMaxAndMin: string;
}

export const WeatherCard = ({ city, hour, weather, temperature, tempMaxAndMin }: Props) => {
  return (
    <StyledContainer>
      <StyledLeftContent>
        <StyledCityContent>
          <StyledTitle>{city}</StyledTitle>
          <StyledSubtitle>{hour}</StyledSubtitle>
        </StyledCityContent>
        <StyledSubtitle>{weather}</StyledSubtitle>
      </StyledLeftContent>
      <StyledRightContent>
        <StyledTemperature>{temperature}</StyledTemperature>
        <StyledSubtitle>{tempMaxAndMin}</StyledSubtitle>
      </StyledRightContent>
    </StyledContainer>
  );
}
