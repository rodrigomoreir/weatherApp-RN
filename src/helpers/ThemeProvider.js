import React from 'react';
import { ThemeProvider as DefaultThemeProvider } from 'styled-components/native';
import PropTypes from 'prop-types';

import { defaultTheme } from '../global/styles/theme';


const ThemeProvider = ({ children }) => (
  <DefaultThemeProvider theme={defaultTheme}>{children}</DefaultThemeProvider>
);

ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired
};

export default ThemeProvider;
