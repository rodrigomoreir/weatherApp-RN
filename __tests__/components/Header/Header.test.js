import React from 'react'

import { renderWithTheme, matchSnapshotWithTheme } from '../../../src/helpers/testing'

import { Header } from '../../../src/components/Header'

describe('Header', () => {
  test('matches snapshot', () => {
    const onPressMock = jest.fn()
    matchSnapshotWithTheme(<Header title={'Header Test'} goBack={onPressMock} />)
  })

  test('renders correctly', () => {
    const onPressMock = jest.fn()
    const { getByText } = renderWithTheme(
      <Header title={'Header Test'} goBack={onPressMock} />,
    )
    const header = getByText('Header Test')
    expect(header).toBeTruthy()
  })

  test('render header with goBack', () => {
    const goBackMock = jest.fn()
    renderWithTheme(
      <Header
        goBack={goBackMock}
        title={'Header Test'}
      />,
    )
  })
})
