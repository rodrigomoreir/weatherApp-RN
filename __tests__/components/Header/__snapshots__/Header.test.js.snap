// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F8FA",
          "borderColor": "#808080",
          "borderRadius": 20,
          "borderWidth": 0.5,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/icons/chevronLeft.png",
          }
        }
        style={
          Array [
            Object {
              "height": 20,
              "tintColor": "#808080",
              "width": 20,
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 25,
            "fontWeight": "bold",
            "marginLeft": 20,
          },
        ]
      }
    >
      Header Test
    </Text>
  </View>
</View>
`;
